core/bones/password.py

```diff
@@ -69,21 +63,58 @@ class PasswordBone(StringBone):
         "core.bones.password.tooWeakMessage",
         defaultText="The entered password is too weak."
     )
+    _password_tests: List[Tuple] = [
+        (r"(?=.*[A-Z])", translate("core.bones.password.no_capital_letters",
+                                   defaultText="The password entered has no capital letters.")),
+        (r"(?=.*[a-z])", translate("core.bones.password.no_lowercase_letters",
+                                   defaultText="The password entered has no lowercase letters.")),
+        (r"(?=.*\d)", translate("core.bones.password.no_digits",
+                                defaultText="The password entered has no digits.")),
+        (r"(?=.*\W)", translate("core.bones.password.no_special_characters",
+                                defaultText="The password entered has no special characters.")),
+
+    ]
+
+    def __init__(
+        self,
+        *,
+        min_password_length: int = 8,
+        test_threshold: int = 3,
+        password_tests: List[Tuple] = None,
+        **kwargs
+    ):
+        """
+            Initializes a new Password Bone.
+
+            :param min_password_length: The minimum length of the password all passwords with a length that is
+                    smaller this will be invalid.
+
+            :param test_threshold: The minimum number of tests the password must pass.
+            :param password_tests: A list of tuple.The tuple  contains the tests and an error message if the test fails
+        """
+        super().__init__()
+        self.min_password_length = min_password_length
+        self.test_threshold = test_threshold
+        self.password_tests = self._password_tests if password_tests is None else password_tests
 
     def isInvalid(self, value):
         if not value:
             return False
 
-        if len(value) < self.minPasswordLength:
-            return self.tooShortMessage.translate(length=self.minPasswordLength)
+        if len(value) < self.min_password_length:
+            return self.tooShortMessage.translate(length=self.min_password_length)
 
         # Run our password test suite
-        testResults = []
-        for test in self.passwordTests:
-            testResults.append(test(value))
-
-        if sum(testResults) < self.passwordTestThreshold:
-            return str(self.tooWeakMessage)
+        tests_errors = []
+        tests_passed = 0
+        for test in self.password_tests:
+            if re.search(test[0], value):
+                tests_passed += 1
+            else:
+                tests_errors.append(str(test[1]))  # we may need to convert a "translate" object
```

Comment:
T think tuple unpacking is more readable and pythonic.

```suggestion
        for test, hint in self.password_tests:
            if re.search(test, value):
                tests_passed += 1
            else:
                tests_errors.append(str(hint))  # we may need to convert a "translate" object
```